package/src/components/ProgressControl/ProgressControl.tsx

```diff
@@ -67,51 +74,43 @@ const ProgressControlThumb = () => {
 
 export const ProgressControl = (props: ProgressControlProps) => {
   const [widthInNumbers, setWidthInNumbers] = useState(0);
-  const {
-    filledColor: filledColorFromProp,
-    onEndDrag,
-    onPlayPause,
-    onProgressDrag,
-    onStartDrag,
-    progress,
-    testID,
-  } = props;
+  const { filledColor: filledColorFromProp, onEndDrag, onStartDrag, progress, testID } = props;
 
-  const progressValue = useSharedValue(progress);
+  const state = useSharedValue(progress);
+  const isSliding = useRef(false);
   const {
     theme: {
       colors: { grey_dark },
       progressControl: { container, filledColor: filledColorFromTheme, filledStyles, thumb },
     },
   } = useTheme();
 
+  useAnimatedReaction(
+    () => progress,
+    (newProgress) => {
+      if (!isSliding.current) {
+        state.value = newProgress;
+      }
+    },
+    [progress, isSliding.current],
+  );
+
   const pan = Gesture.Pan()
```

Comment:
Please memoize the `Pan` gesture. Otherwise, the native controllers are going to be recreated every time anything changes here.

Reply:
Done